### PR TITLE
texlive: clean up recipe

### PR DESCRIPTION
**Pull request by jmcarcell**
Remove the live version and other older versions that were deprecated (some only in the comments) around 2020 and 2021. After that, several parts of the recipe can be simplified. In addition, remove the strict requirement on `poppler` that was introduced several years ago. I have been able to build texlive (2022) with the most recent version of `poppler` so I assume the requirement has not been valid for a while.

### CODE REVIEW

File: var/spack/repos/builtin/packages/texlive/package.py [hunk 1]
```diff
@@ -86,52 +71,34 @@ class Texlive(AutotoolsPackage):
...
+    depends_on("libgd")
+    depends_on("libpaper")
+    depends_on("libpng")
+    depends_on("libxaw")
+    depends_on("libxt")
+    depends_on("mpfr")
+    depends_on("perl")
+    depends_on("pixman")
+    depends_on("poppler@:0.84", when="@:2019")
+    depends_on("poppler")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/texlive/package.py]: 2020 may be the last version that actually depends on poppler, see https://github.com/TeX-Live/texlive-source/blob/e888cd260121c19da9d6f294c9e03f35c05b40d3/m4/kpse-xpdf-flags.m4#L15 and https://github.com/TeX-Live/texlive-source/commit/ef05ec36803a62e5557ee2c6d107802ad544f193.

File: var/spack/repos/builtin/packages/texlive/package.py [hunk 2]
```diff
@@ -86,52 +71,34 @@ class Texlive(AutotoolsPackage):
...
+    depends_on("icu4c")
+    depends_on("libgd")
+    depends_on("libpaper")
+    depends_on("libpng")
+    depends_on("libxaw")
+    depends_on("libxt")
+    depends_on("mpfr")
+    depends_on("perl")
+    depends_on("pixman")
+    depends_on("poppler@:0.84", when="@:2019")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/texlive/package.py]: 0.84 is actually the first version to contain the C++ headers, https://gitlab.freedesktop.org/poppler/poppler/-/commit/1c1d9a70013125f67518f398a2b5abf37ec43cdd, so this seems to have been wrong for a while.
```suggestion
    depends_on("poppler@:0.83", when="@:2019")
```

**Review by wdconinc — COMMENTED** [review]
Thanks for the cleanup. Poppler appears to not be a dependency anymore in current versions of texlive.

I do wonder if it is necessary to remove the legacy version detection for external installs. Those can be useful for people in those systems and is unrelated to maintenance burden of the `live` version.

File: var/spack/repos/builtin/packages/texlive/package.py (hunk 2 above)
Comment (jmcarcell) [reply var/spack/repos/builtin/packages/texlive/package.py]: Change to 0.83

File: var/spack/repos/builtin/packages/texlive/package.py (hunk 1 above)
Comment (jmcarcell) [reply var/spack/repos/builtin/packages/texlive/package.py]: In my local installation (from the system) poppler is not needed so this seems to be true. I changed it to only apply until 2020.

**Comment by jmcarcell** [comment]
> I do wonder if it is necessary to remove the legacy version detection for external installs. Those can be useful for people in those systems and is unrelated to maintenance burden of the `live` version.

Do you mean https://github.com/spack/spack/pull/45863/files#diff-784b155faa9961b422728d82ff6e82dffa19268983c0ab574c550abc4481c70bL237-L245? I removed them because even the author said these were old versions.

**Review by wdconinc — APPROVED** [review]
Thanks for the cleanup. Glad we can remove the poppler dependency.

I think I agree that we can remove the external find support for ubuntu18.04, centos7 and centos8. All are EOL.

**Pull request merged**